module/zcommon/zfs_prop.c

```diff
@@ -680,6 +727,37 @@ zfs_prop_inheritable(zfs_prop_t prop)
 	    zfs_prop_table[prop].pd_attr == PROP_ONETIME);
 }
 
+/*
+ * Returns TRUE if property is one of the encryption properties that requires
+ * a loaded encryption key to modify.
+ */
+boolean_t
+zfs_prop_encryption_key_param(zfs_prop_t prop)
+{
+	/*
+	 * keylocation does not count as an encryption property. It can be
+	 * changed at will without needing the master keys.
+	 */
+	return (prop == ZFS_PROP_PBKDF2_SALT || prop == ZFS_PROP_PBKDF2_ITERS ||
+	    prop == ZFS_PROP_KEYFORMAT);
+}
+
+/*
+ * Helper function used by both kernelspace and userspace to check the
+ * keylpcation property.
+ */
+boolean_t
+zfs_prop_valid_keylocation(const char *str)
+{
+	if (strlen(str) == 6 && strncmp("prompt", str, 6) == 0)
+		return (B_TRUE);
+	else if (strlen(str) > 8 && strncmp("file:///", str, 8) == 0)
```

Comment:
This function is something I was trying to think of ways to improve or possibly remove. Realistically, `zfs_prop_valid_keylocation()` is just a sanity check. The `create`, `clone`, and `change-key` code all will actually need to load the key from the given `keylocation` before they can even attempt to talk to the kernel. This is all done in userspace, however, so the kernel can't really be sure if the `keylocation` it receives is valid. As a result, I added this function as a quick smoke test for the kernel. Can you think of a better way to check this?